packages/wallet/src/utils/transaction-generator.ts

```diff
@@ -126,3 +138,22 @@ export function createDepositTransaction(destination: string, depositAmount: str
   }
   return createNitroDepositTransaction(destination, expectedHeld, depositAmount);
 }
+
+export function createTransferAllTransaction(source: string, destination: string, amount: string) {
+  const allocation: Allocation = [
+    {
+      destination: normalizeDestinationAddress(destination),
+      amount
+    } as AllocationItem
+  ];
+  return createNitroTransferAllTransaction(source, allocation);
+}
+
+function normalizeDestinationAddress(address: string): string {
+  let normalizedDestinationAddress = address;
+  // If the address is not already left-padded to be of type byte32
+  if (normalizedDestinationAddress.length !== 66) {
+    normalizedDestinationAddress = convertAddressToBytes32(address);
+  }
+  return normalizedDestinationAddress;
```

Comment:
```suggestion
  return address.length === 66 ? address : convertAddressToBytes32(address);
```

Reply:
The suggested change feature only worked on one line, not all of them, so you'll have to accept this suggestion manually.

Reply:
Yeah that's why I didn't commit the suggestion